source/clear-linux/guides/maintenance/cpu-performance.rst

```diff
@@ -0,0 +1,165 @@
+
+   .. code::
+
+      sudo systemctl mask clr-power.timer
+
+
+#. Change the governor. In the example below, the governor is set to
+   *performance*.
+
+   .. code::
+
+      echo performance | sudo tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor 
+   
+
+The list of all governors can be found in the `Linux kernel documentation on
+CPUFreq Governors`_.
+
+.. note::
+
+  The intel_pstate driver only supports *performance* and *powersave* governors.
+
+
+Thermal management
+==================
+
+`thermald`_ is a Linux thermal management daemon used to prevent the
+overheating of platforms. It monitors temperature and controls available
+cooling methods accordingly.
+
+By default, thermald is disabled in |CL|. thermald can be enabled using the
+systemd service by running the command: 
+
+.. code::
+   
+   sudo systemctl enable thermald
+
+
+For more information, see the thermald man page :command:`man thermald`.
+
+
+
+Reference topics
+================
+
+C-states (Idle states)
+----------------------
+
+C-states are hardware sleep states entered when the CPU is determined to be
+idle and not executing instructions. 
+
+C-states aim to reduce power utilization by increasingly reducing clock
+frequency, voltages, and features in each state. The tradeoff is a slightly
+longer time to exit the idle state when the CPU is needed once again.
+
+Although C-states can typically be limited or disabled in a system's BIOS or
+EFI configuration, these settings are overidden when the `intel_idle driver`_
+is in use.
+
+For troubleshooting, C-states can be limited with a kernel commandline boot
+parameter by adding :command:`processor.max_cstate=N intel_idle.max_cstate=N`
+or completely disabled with :command:`idle=poll`.
+
+.. note::
+
+   * :commands:`processor.max_cstate=0` is changed to :command:`processor.max_cstate=1` by the kernel
+     to be a valid value.
+   
+   * :command:`intel_idle.max_cstate=0` disables the Intel Idle driver, not set it to
+     C-state 0.
+
+
+
+P-states (Performance states)
+-----------------------------
+
+P-states, also known as *SpeedStep* on Intel processors or *Cool'n'Quiet* on
+AMD processors, are states entered while the CPU is active and executing
+instructions.
+
+P-states aim to reduce power utilization by adjusting CPU clock frequency and
+voltages based on CPU demand.
+
+P-states can typically be limited or disabled in a system's firmware (BIOS/EFI).
+
+Turbo boost
+~~~~~~~~~~~
+
+Turbo boost, found on some modern Intel CPUs, allows core(s) on a processor to
+temporarily operate at a higher than rated CPU clock frequency to accomodate
+demanding workloads if the CPU is under defined power and thernal thresholds. 
+
+Turbo boost is an extension of P-states. As such, changing or limiting C-states
+or P-states impact the ability of a process to enter Turbo boost. 
+
+Turbo boost can be disabled in a system's EFI or BIOS. Turbo boost can also be
```

Comment:
just a note...EFI normally refers to the Intel-specific EFI 1.x BIOS architecture, UEFI refers to the unified 2.x+ version that has been adopted by virtually everyone.  Recommend using UEFI when talking about the UEFI firmware